lib/graph/Workspace.js

```diff
@@ -0,0 +1,255 @@
+import fs from "graceful-fs";
+import {globby, isDynamicPattern} from "globby";
+import path from "node:path";
+import {promisify} from "node:util";
+import {getLogger} from "@ui5/logger";
+import Module from "./Module.js";
+import {validateWorkspace} from "../validation/validator.js";
+
+const readFile = promisify(fs.readFile);
+const log = getLogger("graph:Workspace");
+
+
+/**
+ * Dependency graph node representing a module
+ *
+ * @public
+ * @typedef {object} @ui5/project/graph/Workspace~Configuration
+ * @property {string} node.specVersion
+ * @property {object} node.metadata Version of the project
+ * @property {string} node.metadata.name Name of the workspace configuration
+ * @property {object} node.dependencyManagement
+ * @property {@ui5/project/graph/Workspace~DependencyManagementResolutions[]} node.dependencyManagement.resolutions
+ */
+
+/**
+ * A resolution entry for the dependency management section of the configuration
+ *
+ * @public
+ * @typedef {object} @ui5/project/graph/Workspace~DependencyManagementResolution
+ * @property {string} path Relative path to use for the workspace resolution process
+ */
+
+/**
+ * Workspace representation
+ *
+ * @public
+ * @class
+ * @alias @ui5/project/graph/Workspace
+ */
+class Workspace {
+	#visitedNodePaths = new Set();
+	#configValidated = false;
+	#configuration;
+	#cwd;
+
+	/**
+	 * @public
+	 * @param {object} options
+	 * @param {string} options.cwd Path to use for resolving all paths of the workspace configuration from.
+	 *   This should contain platform-specific path separators (i.e. must not be POSIX on non-POSIX systems)
```

Comment:
```suggestion
	 *  This should contain platform-specific path separators (i.e. must not be POSIX on non-POSIX systems)
```

Reply:
Recently I've used this style (three spaces) but we should really try to come up with a guideline for this 😓

Reply:
True... and maybe we can enforce this with an eslint rule.